src/reflect/scala/reflect/io/ZipArchive.scala

```diff
@@ -112,7 +112,7 @@ abstract class ZipArchive(override val file: JFile, release: Option[String]) ext
   private def ensureDir(dirs: java.util.Map[String, DirEntry], path: String, zipEntry: ZipEntry): DirEntry = {
     dirs get path match {
       case null =>
-        val parent = ensureDir(dirs, dirName(path), null)
+        val parent = ensureDir(dirs, dirNameUsingLast(path), null)
```

Comment:
Not sure this is useful. This is recursive, seeing successively shorter paths until root or some parent already is in dirs map. Also, it zaps the cached value for the first leaf, so second leaf recalculates; for getdir("/a/b"), getdir("/a/c").